### PR TITLE
Don't crash if session.state is {}

### LINKED ISSUE #2848: TypeError: Cannot read property 'show_sidebar_tutorial' of undefined

### PR DESCRIPTION
**Pull request by seanh**
Fixes #2848.

### CODE REVIEW

**Comment by nickstenning** [comment]
This should probably have a test :)

**Comment by seanh** [comment]
Tests added

File: h/static/scripts/directive/sidebar-tutorial.js [hunk 1]
```diff
@@ -1,12 +1,17 @@
...
 // @ngInject
-function controller(session) {
+function SidebarTutorialController(session) {
   /*jshint validthis:true */
   var vm = this;
 
   vm.showSidebarTutorial = function () {
-    return session.state.preferences.show_sidebar_tutorial;
+    if (session.state.preferences) {
+      if (session.state.preferences.show_sidebar_tutorial === true) {
```
Comment (robertknight) [inline h/static/scripts/directive/sidebar-tutorial.js]: In keeping with our current coding conventions, this should just be `if (session.state.preferences.show_sidebar_tutorial)`

Comment (seanh) [inline h/static/scripts/directive/sidebar-tutorial.js]: Done

**Pull request merged**